src/adapt.c

```diff
@@ -54,6 +54,21 @@
     return ENOMEM; \
 }
 
+#define KNOBS_LOOP(_array, _index, _switch, _status, _cpu) \
```

Comment:
Oh, I didn't look look right of the types because we use a pointer in the structure and not the structure itself. So I make a function to perform this action.